File: apprefurbished/src/main/java/com/androidcollider/easyfin/models/Transaction.java
package com.androidcollider.easyfin.models;

import java.io.Serializable;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * @author Ihor Bilous
 */

@Getter
@Builder
@ToString
@EqualsAndHashCode(exclude = {"idAccount", "category", "accountType", "date", "amount", "accountAmount", "currency", "accountName"})
public class Transaction implements Serializable {

    @Setter
    private int id;
    private int idAccount;
    private int category;
    private int accountType;
    private long date;
    private double amount;
    private double accountAmount;
    private String currency;
    private String accountName;
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/models/Transaction.kt
package com.androidcollider.easyfin.common.models

import java.io.Serializable

/**
 * @author Ihor Bilous
 */
class Transaction : Serializable {
    var id = 0
    var idAccount = 0
    var category = 0
    var accountType = 0
    var date: Long = 0
    var amount = 0.0
    var accountAmount = 0.0
    var currency: String? = null
    var accountName: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Transaction
        if (id != other.id) return false
        return true
    }

    override fun hashCode(): Int {
        return id
    }
}